libsolidity/formal/CHC.cpp

```diff
@@ -60,39 +60,42 @@ using namespace solidity::frontend::smt;
 CHC::CHC(
 	EncodingContext& _context,
 	UniqueErrorReporter& _errorReporter,
-	[[maybe_unused]] map<util::h256, string> const& _smtlib2Responses,
-	[[maybe_unused]] ReadCallback::Callback const& _smtCallback,
+	map<util::h256, string> const& _smtlib2Responses,
+	ReadCallback::Callback const& _smtCallback,
 	ModelCheckerSettings const& _settings,
 	CharStreamProvider const& _charStreamProvider
 ):
+	m_smtlib2Responses(_smtlib2Responses),
+	m_smtCallback(_smtCallback),
 	SMTEncoder(_context, _settings, _errorReporter, _charStreamProvider)
```

Comment:
```suggestion
	SMTEncoder(_context, _settings, _errorReporter, _charStreamProvider),
	m_smtlib2Responses(_smtlib2Responses),
	m_smtCallback(_smtCallback)
```
should fix the build

Reply:
Or at least half of it... may also have some forward-declaration issues apart from the ordering here...